### PR TITLE
Add Get/Set Text Record, Extend Domain Support

### PR DESCRIPTION
**Pull request by nitinshr135**
Fixes SPH-1709 SPH-1711 SPH-1712 SPH-1713 SPH-1714 SPH-1718 SPH-1720 SPH-1721 SPH-1722 SPH-1723

### CODE REVIEW

**Review by SingingApple — CHANGES_REQUESTED** [review]
Let's have interfaces for all responses now if feasible

File: packages/frontend/src/components/InputField/edit-subdomain.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  subdomainName,
+  editSubdomainName,
+}: IProps) {
+  const [inputFocus, setInputFocus] = useState<boolean>(false)
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const handleEnterPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter') {
+      // handleSearch()
```
[the remark this answers is not in the document]
Comment (abhinav-spheron) [reply packages/frontend/src/components/InputField/edit-subdomain.tsx:24]: @nitinshr135 Check this

File: packages/frontend/src/components/InputField/edit-subdomain.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import React, { useRef, useState } from 'react'
+import { ReactComponent as GlobeLogo } from '../../assets/icons/globe-icon.svg'
+import { Input } from './input'
+
+interface IProps {
+  classname: string
+  domainName: string
+  subdomainName: string
+  editSubdomainName: (e: any) => void
```
Comment (abhinav-spheron) [inline packages/frontend/src/components/InputField/edit-subdomain.tsx:9]: Add appropriate type

**Review by abhigaur12 — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/InputField/edit-subdomain.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (nitinshr135) [reply packages/frontend/src/components/InputField/edit-subdomain.tsx:24]: Edit subdomain feature is not yet functional as it is not implemented from the package side.

[the remark this answers is not in the document]
Comment (nitinshr135) [reply packages/frontend/src/components/InputField/edit-subdomain.tsx:24]: Edit subdomain feature is not yet functional as it is not implemented from the package side.

[the remark this answers is not in the document]
Comment (abhinav-spheron) [reply packages/frontend/src/components/InputField/edit-subdomain.tsx:24]: Okay, got it.

**Review by abhigaur12 — APPROVED** [review]
LGTM

**Review by SingingApple — CHANGES_REQUESTED** [review]
A few changes

**Review by rekpero — CHANGES_REQUESTED** [review]
small change needed, otherwise lgtm

File: packages/frontend/src/components/Modal/extend-modal.tsx [hunk 3]
```diff
@@ -1,110 +1,219 @@
...
+    getGasFee()
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [params.domainName, year, option.isDomainAvailable])
+
+  const handleExtend = async () => {
+    setExtendLoading(true)
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      const res: any = await renewNames(option.searchQuery, year, price)
```
[the remark this answers is not in the document]
Comment (nitinshr135) [reply packages/frontend/src/components/Modal/extend-modal.tsx:86]: the response is coming as 
image

and the promise if of type `{
    error: boolean;
    response: {
        base: BigNumber;
        premium: BigNumber;
    } | undefined;
} | {
    error: boolean;
    response: string;
}`

getting some or other error while resolving this

[the remark this answers is not in the document]
Comment (nitinshr135) [reply packages/frontend/src/components/Modal/extend-modal.tsx:86]: cc @rekpero

**Review by rekpero — APPROVED** [review]

File: packages/frontend/src/components/Modal/extend-modal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (nitinshr135) [reply packages/frontend/src/components/Modal/extend-modal.tsx:86]: yes can be done 
cc @rekpero

**Review by SingingApple — APPROVED** [review]

**Pull request merged**